File: java/proxy/proxy-video/src/main/java/uea/edu/ParentalControlProxy.java
package uea.edu;

// Proxy para controle parental
public class ParentalControlProxy implements VideoService {
    private RealVideoService realVideoService = new RealVideoService();

    public void playVideo(String videoTitle) {
        if (isRestricted(videoTitle)) {
            System.out.println("Este vídeo está bloqueado pelo controle parental: " + videoTitle);
        } else {
            realVideoService.playVideo(videoTitle);
        }
    }

    // Método para simular restrições de conteúdo
    private boolean isRestricted(String videoTitle) {
        // Apenas um exemplo; em uma aplicação real, poderia haver uma lista de verificação de restrições
        return videoTitle.toLowerCase().contains("violência") || videoTitle.toLowerCase().contains("adulto");
    }

}

File: python/proxy/proxy-video/ParentalControlProxy.py
from VideoService import VideoService
from RealVideoService import RealVideoService

# Proxy para controle parental
class ParentalControlProxy(VideoService):
    def __init__(self):
        self._real_video_service = RealVideoService()

    def play_video(self, video_title):
        if self.is_restricted(video_title):
            print(f"Este vídeo está bloqueado pelo controle parental: {video_title}")
        else:
            self._real_video_service.play_video(video_title)

    # Método para simular restrições de conteúdo
    def is_restricted(self, video_title):
        # Verificação de restrição simples; pode ser expandida
        restricted_keywords = ["violência", "adulto"]
        return any(keyword in video_title.lower() for keyword in restricted_keywords)